src/Mustache/Loader/FilesystemLoader.php

```diff
@@ -96,25 +100,29 @@ public function load($name)
      */
     protected function loadFile($name)
     {
-        $fileName = $this->getFileName($name);
+        foreach ($this->baseDir as $dir) {
+            $fileName = $this->getFileName($name, $dir);
 
-        if ($this->shouldCheckPath() && !file_exists($fileName)) {
-            throw new Mustache_Exception_UnknownTemplateException($name);
+            if (!$this->shouldCheckPath($dir) || file_exists($fileName)) {
+                return file_get_contents($fileName);
```

Comment:
this is a little bit weird, because if any "directory" in `$this->baseDir` uses a stream loader, it will never even check any directory after it in the list.